pwr.md

```diff
@@ -0,0 +1,147 @@
+
+```python
+core.min_freq = core.lowest_freq  # Set the desired minimum frequency to be lowest available
+core.max_freq = core.highest_freq  # Set the desired maximum frequency to the highest available
+
+cpu.uncore_max_freq = cpu.uncore_hw_max  # Set the desired maximum uncore frequency to the highest available
+cpu.uncore_min_freq = cpu.uncore_hw_min  # Set the desired minimum uncore frequency to the lowest available
+```
+
+### Committing 
+Modification of the power settings of a system is done by altering the core or CPU characteristics, as shown above, 
+then finalizing with the `commit()` function call.
+```python 
+cores = pwr.get_cores()  # Create the cores object list 
+
+for core in cores:  # Loop through core objects in list
+    core.min_freq = core.base_freq  # Set the desired minimum frequency to be the base frequency 
+    core.max_freq = core.highest_freq  # Set the desired maximum frequency to the highest available
+    core.commit()  # Commit changes to be made on system
+```
+
+
+### Pre-set Profiles
+When an application is modifying the desired min and max core frequencies, pre-set configurations can also be applied, these will overwrite current configurations and commit the pre-sets.  
+* `minimum`:            Set all cores minimum to 800Mhz and maximum to 800Mhz.
+* `maximum`:            Set all cores minimum to 3900Mhz and maximum to 3900Mhz.
+* `base`:               Set all cores minimum to 2300Mhz and maximum to 2300Mhz.
+* `default`:            Set all cores minimum to 800Mhz and maximum to 3900Mhz.
+* `no_turbo`:           Set all cores minimum to 800Mhz and maximum to 2300Mhz.
+* `sst_bf_base`:        Set high priority cores minimum and maximum to 2700Mhz, normal priority cores minimum and maximum to 2100Mhz (Only available with SST-BF).
+* `sst_bf_high_turbo`:  Set high priority cores minimum to 2700Mhz and maximum to 3900Mhz, normal priority cores minimum to 800Mhz and maximum to 2100Mhz(Only available with SST-BF).
+* `sst_bf_low_turbo`:   Set high priority cores minimum to 2700Mhz and maximum to 2700Mhz, normal priority cores minimum to 800Mhz and maximum to 3900Mhz(Only available with SST-BF).
+>Specific frequencies will depend on system and configuration.
+
+```python
+For c in cores:
+    c.commit("sst_bf_base") # Configure cores with the SST-BF configuration
+```
+
+## Refreshing cpu stats
+CPU stats can become out of date, such as `curr_freq` or `sst_bf_configured`. These can be refreshed in with `refresh_stats()`.
+It is *advised* that CPU and core objects be updated with `refresh_stats()` before requesting object data.
+
+core.refresh_stats() will update:
+* `curr_freq`             
+* `min_freq`                
+* `max_freq`                
+* `epp`
+                
+cpu.refresh_stats() will update:
+* `uncore_freq`
+* `uncore_max_freq` 
+* `uncore_min_freq`
+* `sst_bf_configured`
+
+```python
+for c in cores:
+    c.refresh_stats()
+    if not c.sst_bf_enabled:
```

Comment:
still not sure of this code. if SST-BF is not enabled, why would you commit SST-BF-related profile? i think you meant configured?

Reply:
yes, mix-up.